### PR TITLE
Add --createdb to create the MySQL database. 

### PR DESCRIPTION
**Pull request by philip**
Adds new `-c,--createdb` command line option to create the MySQL database. Also introduces new DB_* configuration options (`DB_HOST`, `DB_USERNAME`, `DB_PASSWORD`, and `DB_PORT`) that default to the current lambo defaults. They can be edited via `edit-config`. In addition, these values are used in the generated `.env` file.

### CODE REVIEW

**Comment by philip** [comment]
All thoughts and suggestions are welcome. :)

**Comment by philip** [comment]
I'm new to Laravel, and a Bash novice, so feel free to dismiss or edit this PR without any worry. It also might make sense doing this a completely different way (e.g., in PHP).

File: lambo [hunk 1]
```diff
@@ -298,6 +309,30 @@ if [[ "$NODE" = true ]]; then
...
+# Defining MYSQL_PWD to avoid insecure password warning... user beware
+if [[ "$CREATEDB" != "" ]]; then
+    if [[ "$CREATEDB" == "sqlite" ]]; then
+        echo "${green}Creating an SQLite database here: database/$PROJECTNAME.sqlite${reset}"
+        DB_CONNECTION="sqlite"
+        DB_NAME="database\/$PROJECTNAME.sqlite"
+        touch "database/$PROJECTNAME.sqlite"
+    elif [[ "$CREATEDB" == "mysql" ]]; then
+        echo "${green}Checking if a MySQL database named $PROJECTNAME already exists${reset}"
+        DBEXISTS=`MYSQL_PWD=$DB_PASSWORD mysql -u$DB_USERNAME -P$DB_PORT -h$DB_HOST -e "SHOW DATABASES" | grep $PROJECTNAME`
```
Comment (mattstauffer) [inline lambo]: Is there any other safe way to pass the command in without getting warnings?

Or could we possibly unset the env var afterward at least?

Comment (mattstauffer) [reply lambo]: Should this grep `$DB_NAME` instead of `$PROJECTNAME`?

Comment (philip) [reply lambo]: AFAIK this only sets `MYSQL_PWD` temporarily so it won't persist as it's not exported. At least, neither `printenv` nor `set` reveal it in lambo after this block.

Agreed, security is an issue here but I'm unaware of a better way that doesn't prompt for the password. We could prompt, though. Or use an API (e.g., PHP->PDO_MYSQL) and read the lambo configuration file. Setting a temporary file and using MySQL's `--defaults-extra-file=foo.cnf` is also an option but I don't feel this complexity is worth it. IMHO using `MYSQL_PWD` is okay here.

Whatever the case, I could write something up along the lines of "Using `--createdb` is a security risk for reasons X and Y" although I suspect this isn't a real-world lambo concern considering empty MySQL passwords are common in local setups.

Good catch, `$DB_NAME` was added but not fully implemented in the second commit, but is now.

Remaining question: for sqlite, many DB_ definitions (e.g., `DB_PASSWORD`) are set in `.env` but not used. Should we remove or set them to ""? Or leave as us to make switching easier later on?

Comment (mattstauffer) [reply lambo]: All makes sense! Let's just leave them for SQLite for later usage.

And yah... can you just add the warning and then we'll merge this? Thanks so much for your work on it--I'll use this flag often!

Comment (philip) [reply lambo]: Oh my, this PR is getting out of control :) 

While attempting to merge in the perl changes from the other PR, I notice this all introduces complexities. For example, if a password contains special characters then it'll break the expression... probably the same is true for sed. Or if the project name begins with a number then it's going to break the database name replacement due to my misguided use of back references.

This is your punishment for trusting me, and mine for thinking this simple desire would be easy to implement for the masses. I'll focus on getting this working the best I can and then lambo can go worry about edge cases after.

**Comment by philip** [comment]
To anyone lurking: dealing with bash and perl got messy so the new lampho branch was born. In other words, lambo is being rewritten in PHP.

For this reason I'm closing this PR.

**Pull request closed without merging**